mesonbuild/utils/universal.py

```diff
@@ -1126,6 +1127,28 @@ def determine_worker_count(varnames: T.Optional[T.List[str]] = None) -> int:
             num_workers = 1
     return num_workers
 
+def path_starts_with(name: str, path: str) -> bool:
+    '''Checks if @name is a file under the directory @path. Both @name and @path should be
+       adequately normalized, and either both or none should be absolute.'''
```

Comment:
I wonder if there is a better name we could use besides name/path?

Reply:
Another function already in meson uses parent/trial. I can switch if preferred.

Reply:
That sounds better, yes.